app/models/vm_or_template.rb

```diff
@@ -174,9 +174,11 @@ class VmOrTemplate < ApplicationRecord
   virtual_has_many  :file_shares,          :class_name => 'SniaFileShare'
   virtual_has_many  :storage_volumes,      :class_name => 'CimStorageVolume'
   virtual_has_many  :logical_disks,        :class_name => 'CimLogicalDisk'
+  virtual_has_many  :children,             :class_name => 'VmOrTemplate'
```

Comment:
This breaks uniq check in `vm.add_children` method at https://github.com/ManageIQ/manageiq/blob/053c16e673d1ba73b98b6ebf73d98f46d2853c75/app/models/mixins/relationship_mixin.rb#L536

You can check yourself by running `Vm.first.add_children(Vm.last)` twice in the row.

That is because the Vm class already has child_ids method (from Relationship mixin) and `virtual_has_many` redefines this method (`child_ids`) with slightly different return value.

Not sure if this is actually useful code in the context of vm relationships.